Sources/UI/Chat/Cells/MessageTableViewCell+Attachments.swift

```diff
@@ -56,18 +56,18 @@ extension MessageTableViewCell {
         messageStackView.insertArrangedSubview(preview, at: index)
         attachmentPreviews.append(preview)
         
-        // File preview.
         guard let imagePreview = preview as? ImageAttachmentPreview else {
-            preview.update { _, _ in }
-            preview.update(imageMask: backgroundImageForAttachment(at: index))
+            // File preview.
+            (preview as? FileAttachmentPreview)?.update()
```

Comment:
Why don't you make the parameter optional in protocol so you don't need to cast?

Reply:
I can't provide a default value anyway. it will be: `update(nil)`

Reply:
Do you think an optional completion block is better?

Reply:
```swift
protocol A {
    func update(_ completion: () -> ())
}

extension A {
    func update(_ completion: () -> () = { }) {
        update(completion)
    }
}

class B: A {
    func update(_ completion: () -> () = { }) {
        // ...
    }
}
```
better if it's optional

Reply:
> Do you think an optional completion block is better?
Definitely for this case